src/stages/models/helper.ts

```diff
@@ -27,3 +38,53 @@ export function convertPrismaToStageResponse(prismaObjects: StagePrismaObject):
 export function convertArrayPrismaStageToStageResponse(prismaObjects: StagePrismaObject[]): StageModel[] {
   return prismaObjects.map((stage) => convertPrismaToStageResponse(stage));
 }
+
+/**
+ * This function aggregates task statuses for a given stage ID.
+ * @param stageId The ID of the stage to aggregate task statuses for.
+ * @param prisma The Prisma client instance.
+ * @returns A promise that resolves to an object containing the aggregated task statuses.
+ */
+export async function aggregateTaskStatusesForStage(stageId: string, prisma: PrismaClient): Promise<StageSummary> {
+  const taskCounts = await prisma.task.groupBy({
+    by: ['status'],
+    where: {
+      stageId: stageId,
+    },
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    _count: {
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      _all: true,
+    },
+  });
+
+  // Explicitly type the taskCounts array
+  interface TaskCount {
+    status: TaskOperationStatus;
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    _count: { _all: number };
+  }
+
+  const aggregatedCounts = taskCounts.reduce<StageSummary>(
+    (acc, count: TaskCount) => {
+      acc[count.status] = count._count._all;
+      return acc;
+    },
+    { ...defaultStatusCounts }
+  );
+
+  return aggregatedCounts;
+}
+
+/**
+ * This method calculates the current percentage of completed tasks in a stage.
+ * @param stageSummary summary of the stage
+ * @returns percentage of completed tasks
+ */
+export function getCurrentPercentage(stageSummary: StageSummary): number {
```

Comment:
i would really consider also saving total tasks instead of calculating each time. it would also make it easier to create dashboards over the db with grafana.

Reply:
done